src/scripts/modules/media/body/body.coffee

```diff
@@ -152,6 +152,27 @@ define (require) ->
             # NOTE: `.toggleClass()` explicitly requires a `false` (not falsy) 2nd argument
             $el.toggleClass('ui-has-child-title', $title.length > 0)
 
+          # Create subtitles in each chapter with corresponding ids as their href
+          $totalSubtitle = $temp.find("[data-depth='1']")
+          $totalId = $totalSubtitle.children('[data-type="title"]')
+          idLength = $totalId.length
+          countIndex = 0
+          realHref = window.location.pathname
+          #Determine if this is a outer page view or not
+          $location = $(".table-of-contents [href$='#{realHref}'] > .title")
+          # Keep adding subtitles until all subtitles are added
+          if $location.children("li").length < idLength
+            while countIndex < idLength
```

Comment:
I'm curious, why did `.map(...)` not work? See https://github.com/Connexions/webview/pull/1545#discussion_r120741070